load.php

```diff
@@ -162,9 +162,36 @@ function( $module_settings ) {
 	 */
 	$modules = apply_filters( 'perflab_active_modules', $modules );
 
+	if ( ! function_exists( 'is_plugin_active' ) ) {
+		include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+	}
+
+	$standalone_plugin_config = perflab_get_standalone_plugins_config();
+	foreach ( $standalone_plugin_config as $plugin_slug => $plugin_main_file ) {
+		if ( in_array( $plugin_slug, $modules, true ) && is_plugin_active( $plugin_main_file ) ) {
+			$key = array_search( $plugin_slug, $modules, true );
+			if ( false !== $key ) {
+				unset( $modules[ $key ] );
+			}
+		}
+	}
+
```

Comment:
This logic shouldn't be added to this function. Even when a standalone plugin is loaded, the module itself is technically still active, so this function should continue returning it. Maybe we can instead add it to `perflab_can_load_module()`? That would make more sense contextually.

Since we need a way to know _why_ a module cannot be loaded then outside of that function, I would suggest we modify the `@return` type of `perflab_can_load_module()` to return `bool|WP_Error`. This way, we can provide an error code and message for the different reasons that the module cannot be loaded. For example, when `can-load.php` returns `false`, it would be a different reason than when the standalone plugin is already loaded. WDYT?

Reply:
Agreed and updated based on your feedback.